spec/models/mixins/ownership_mixin_spec.rb

```diff
@@ -0,0 +1,76 @@
+describe OwnershipMixin do
+  let(:user) { FactoryGirl.create(:user_with_email_and_group) }
+  let(:other_user) { FactoryGirl.create(:user) }
+  let(:group) { user.current_group }
+  let(:vm) { FactoryGirl.create(:vm) }
+  let(:vm_with_user) { FactoryGirl.create(:vm, :evm_owner => user) }
+  let(:vm_with_group) { FactoryGirl.create(:vm, :miq_group => group) }
+
+  describe ".evm_owner_email" do
+    it "shows blank user" do
+      expect(vm.evm_owner_email).to be_nil
+    end
+
+    it "shows user" do
+      expect(vm_with_user.evm_owner_email).to eq(user.email)
+    end
+  end
+
+  describe ".evm_owner_name" do
+    it "shows blank user" do
+      expect(vm.evm_owner_name).to be_nil
+    end
+
+    it "shows user" do
+      expect(vm_with_user.evm_owner_name).to eq(user.name)
+    end
+  end
+
+  describe ".evm_owner_userid" do
+    it "shows blank user" do
+      expect(vm.evm_owner_userid).to be_nil
+    end
+
+    it "shows user" do
+      expect(vm_with_user.evm_owner_userid).to eq(user.userid)
+    end
+  end
+
+  # multiple user names? / downcase example
+  describe ".owned_by_current_user" do
+    it "matches user" do
+      vm
+      vm_with_user
+      User.with_user(user) do
+        expect(Vm.all.order(:id)
+                 .select(:id, :evm_owner_id, Vm.arel_attribute(:owned_by_current_user).as("owned_by_current_user"))
+                 .map(&:owned_by_current_user)
+              ).to eq([nil, true])
+      end
+    end
+  end
+
+  describe ".owning_ldap_group" do
+    it "shows blank group" do
+      expect(vm.owning_ldap_group).to be_nil
+    end
+
+    it "shows group" do
+      expect(vm_with_group.owning_ldap_group).to eq(group.description)
+    end
+  end
+
+  # multiple groups? / downcase example
+  describe ".owned_by_current_ldap_group" do
+    it "matches user" do
```

Comment:
Seems like this should be more something like `it "has a return value of true, false or nil"`.